app/src/main/java/de/westnordost/streetcomplete/data/meta/CountryInfo.kt

```diff
@@ -0,0 +1,116 @@
+    val hasLivingStreet get() = infos.getFirstNonEmpty { it.hasLivingStreet }!!
+    val hasNoStandingSign get() = infos.getFirstNonEmpty { it.hasNoStandingSign }!!
+    val hasSlowZone get() = infos.getFirstNonEmpty { it.hasSlowZone }!!
+    val isLeftHandTraffic get() = infos.getFirstNonEmpty { it.isLeftHandTraffic }!!
+    val isUsuallyAnyGlassRecyclableInContainers get() = infos.getFirstNonEmpty { it.isUsuallyAnyGlassRecyclableInContainers }!!
+    val lengthUnits get() = infos.getFirstNonEmpty { it.lengthUnits }
+    val noParkingSignStyle get() = infos.getFirstNonEmpty { it.noParkingSignStyle }!!
+    val noStoppingSignStyle get() = infos.getFirstNonEmpty { it.noStoppingSignStyle }!!
+    val officialLanguages get() = infos.getFirstNonEmpty { it.officialLanguages }
+    val popularReligions get() = infos.getFirstNonEmpty { it.popularReligions }
+    val regularShoppingDays get() = infos.getFirstNonEmpty { it.regularShoppingDays }!!
+    val roofsAreUsuallyFlat get() = infos.getFirstNonEmpty { it.roofsAreUsuallyFlat }!!
+    val speedUnits get() = infos.getFirstNonEmpty { it.speedUnits }
+    val weightLimitUnits get() = infos.getFirstNonEmpty { it.weightLimitUnits }
+    val workweekDays get() = infos.getFirstNonEmpty { it.workweekDays }!!
+
+    // may be null or empty
+    val additionalStreetsignLanguages get() = infos.getFirstNonEmpty { it.additionalStreetsignLanguages }
+    val additionalValidHousenumberRegex get() = infos.getFirstNonEmpty { it.additionalValidHousenumberRegex }
+    val atmOperators get() = infos.getFirstNonEmpty { it.atmOperators }
+    val chargingStationOperators get() = infos.getFirstNonEmpty { it.chargingStationOperators }
+    val clothesContainerOperators get() = infos.getFirstNonEmpty { it.clothesContainerOperators }
+    val livingStreetSignStyle get() = infos.getFirstNonEmpty { it.livingStreetSignStyle }
+    val mobileCountryCode get() = infos.getFirstNonEmpty { it.mobileCountryCode }
+    val noParkingLineStyle get() = infos.getFirstNonEmpty { it.noParkingLineStyle }
+    val noStandingLineStyle get() = infos.getFirstNonEmpty { it.noStandingLineStyle }
+    val noStandingSignStyle get() = infos.getFirstNonEmpty { it.noStandingSignStyle }
+    val noStoppingLineStyle get() = infos.getFirstNonEmpty { it.noStoppingLineStyle }
+    val orchardProduces get() = infos.getFirstNonEmpty { it.orchardProduces }
+    val popularSports get() = infos.getFirstNonEmpty { it.popularSports }
+    val slowZoneLabelPosition get() = infos.getFirstNonEmpty { it.slowZoneLabelPosition }
+    val slowZoneLabelText get() = infos.getFirstNonEmpty { it.slowZoneLabelText }
+
+    val locale: Locale
+        get() = if (officialLanguages.isEmpty()) {
+            Locale.getDefault()
+        } else {
+            Locale(officialLanguages[0], countryCode)
+        }
+}
+
+private fun <T> List<IncompleteCountryInfo>.getFirstNonEmpty(getValue: (IncompleteCountryInfo) -> T): T {
```

Comment:
Can this function really return `null`. The comment in line 84 claims it does.

Reply:
Yes, `T` can also be substituted with a nullable type.
If `getValue` returns null for all country infos, then the return type will be null.

Reply:
Though this is not relevant anymore with [`8f740dc`](https://github.com/streetcomplete/StreetComplete/pull/3715/commits/8f740dc081a50b52fe7150a7f9e762765c416deb).